test/base/LendingPoolTransferTest.js

```diff
@@ -0,0 +1,98 @@
+const Lenders = artifacts.require("./base/Lenders.sol");
+const LendingPool = artifacts.require("./base/LendingPool.sol");
+
+// TODO Which function is testing this test file Transfer or Repay?
+// Answer - The Repay function which triggers the private tokenTransferFrom function. I can delete this test since the LendingPoolRepayTest does the same 
+contract('LendingPoolRepayTest', function (accounts) {
+    const erc20InterfaceEncoder = new ERC20InterfaceEncoder(web3);
+    const compoundInterfaceEncoder = new CompoundInterfaceEncoder(web3);
+
+    let instance;
+    let zTokenInstance;
+    let daiInstance;
+    let lendersInstance;
+    let interestConsensusInstance;
+    let cTokenInstance;
+    let loansAddress = accounts[0];
+
+    beforeEach('Setup for each test', async () => {
+        zTokenInstance = await Mock.new();
+        daiInstance = await Mock.new();
+        instance = await LendingPool.new();
+        interestConsensusInstance = await Mock.new();
+        cTokenInstance = await Mock.new()
+        const settingsInstance = await Mock.new();
+
+        lendersInstance = await Lenders.new(
+          zTokenInstance.address,
+          instance.address,
+          interestConsensusInstance.address
+        );
+
+        await instance.initialize(
+            zTokenInstance.address,
+            daiInstance.address,
+            lendersInstance.address,
+            loansAddress,
+            cTokenInstance.address,
+            settingsInstance.address,
+        );
+    });
+
+    withData({
+        _1_basic: [accounts[1], loansAddress, true, 10, false, 1, undefined, false],
+        _2_notLoan: [accounts[1], accounts[2], true, 10, false, 1, 'Address is not Loans contract.', true],
+        _3_transferFail: [accounts[1], loansAddress, false, 200, false, 1, "TransferFrom wasn't successful.", true],
+        _4_compoundFail: [accounts[1], loansAddress, true, 10, true, 1, 'COMPOUND_DEPOSIT_ERROR', true],
+    }, function(
+        borrower,
+        sender,
+        transferFrom,
+        amountToRepay,
+        compoundFails,
+        allowance,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('user', 'transfer', 'Should able (or not) to transfer.', mustFail), async function() {
+            // Setup
+            const encodeTransferFrom = erc20InterfaceEncoder.encodeTransferFrom();
+            await daiInstance.givenMethodReturnBool(encodeTransferFrom, transferFrom);
+
+            const mintResponse = compoundFails ? 1 : 0
+            const encodeCompMint = compoundInterfaceEncoder.encodeMint();
+            await cTokenInstance.givenMethodReturnUint(encodeCompMint, mintResponse);
+
+            const encodeAllowance = erc20InterfaceEncoder.encodeAllowance();
+            await daiInstance.givenMethodReturnUint(encodeAllowance, allowance);
+
+            try {
+                // Invocation
+                const result = await instance.repay(amountToRepay, borrower, { from: sender });
```

Comment:
This unit test is testing the  "repay"  function. So, the filename and name (see line #17) should match with the function it is testing.

Reply:
removed